improver/cli/interpret_metadata.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2021 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to interpret the metadata of an IMPROVER output."""
+
+from improver import cli
+
+
+@cli.clizefy
+def process(*cubes: cli.inputcube, verbose=False):
+    """Intepret the metadata of an IMPROVER output into human readable format
+    according to the IMPROVER standard. An optional verbosity flag, if set to
+    True, will specify the source of each interpreted element.
+
+    This tool is intended as an aid to developers in adding and modifying
+    metadata within the code base.
+
+    Args:
+        cubes (iris.cube.Cube or list of iris.cube.Cube):
+            The cube(s) for which the metadata should be interpreted.
+        verbose (bool):
+            Boolean flag to output information about sources of metadata
+            interpretation
+
+    Returns:
+        str:
+            A human-readable interpretation of the cube's metadata is returned
+            to standard out.
```

Comment:
This method doesn't return anything. It prints to stdout. It will Raise errors though if the file is non-conformant, which should be mentioned here I think.

Reply:
I've added a Raises section to the doc-string. This is not something we would normally do (i.e. list all the raises of the plugins that the CLI calls), but as it might well be a fairly typical outcome here I guess it is worth including.